# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/xre/nsAppRunner.cpp
# Commit: 49b5bed3deb4
# Full Hash: 49b5bed3deb4df627fe3286f85be030dc021f614
# Author: Mike Hommey <mh+mozilla@glandium.org>
# Date: 2021-07-16 21:43:02
# Description:
#   Bug 1719237 - Revert bug 1718131. r=stransky
#   
#   It turns out calling gdk_display_close gets us a rematch of bug 1626536,
#   so remove the call that was added in bug 1718131, and adjust valgrind
#   suppressions accordingly.
# ==============================================================================

diff -r ba55f401aea8 -r 49b5bed3deb4 toolkit/xre/nsAppRunner.cpp
--- a/toolkit/xre/nsAppRunner.cpp	Fri Jul 16 08:11:50 2021 +0000
+++ b/toolkit/xre/nsAppRunner.cpp	Fri Jul 16 08:21:49 2021 +0000
@@ -3384,12 +3384,6 @@
   ~XREMain() {
     mScopedXPCOM = nullptr;
     mAppData = nullptr;
-#if defined(MOZ_WIDGET_GTK)
-    if (mGdkDisplay) {
-      gdk_display_close(mGdkDisplay);
-      mGdkDisplay = nullptr;
-    }
-#endif
   }
 
   int XRE_main(int argc, char* argv[], const BootstrapConfig& aConfig);
